mllib/src/main/scala/org/apache/spark/mllib/recommendation/MatrixFactorizationModel.scala

```diff
@@ -75,11 +75,29 @@ class MatrixFactorizationModel @Since("0.8.0") (
     }
   }
 
+  /** Validates the user and return the feature corresponding to the user. */
+  private def validateAndGetUser(user: Int): Array[Double] = {
+    userFeatures.lookup(user) match {
+      case userFeature: Seq[Array[Double]] =>
```

Comment:
Why the pattern match? `.lookup` should already be known to return a `Seq` of the type of the RDD's values which is already known to be `Array[Double]`.

The name of the method isn't really accurate either; it gets a user vector, not user.

```
val vec = userFeatures.lookup(user)
require(vec.nonEmpty, ...)
vec.head
```

How about one method instead of two, that takes the RDD as an arg? might cut down on the code expansion here. I'm on the fence about whether this warrants a new method since there are only two call sites, but it's reasonable.

Reply:
Thank you for the comment. 
I have removed both the methods. Because each method has to give distinct exception, like "user not found" in the first method and "product not found" in the second method. So, it is better to remove the methods, instead of making one.
I have added the validation code inside the methods such as predict, recommendProduct etc.

Reply:
@srowen Please let me know if anymore changes required.